Add citation (#961)
metatensor/metatensor: docs/src/cite.rst

.. _cite:

Cite Metatensor
===============

> If you use metatensor for your work, please read and cite our paper:

Reviewer: ```suggestion
If you use metatensor for your work, please read and cite the corresponding article:
```

- F. Bigi, J.W. Abbott, P. Loche et. al. Metatensor and metatomic: foundational
  libraries for interoperable atomistic machine learning, (2025).
  `https://doi.org/10.48550/arXiv.2508.15704
  <https://doi.org/10.48550/arXiv.2508.15704>`_

In BibTeX format:

.. code-block:: bibtex

    @misc{bigi_metatensor_2025,
    title = {Metatensor and Metatomic: Foundational Libraries for Interoperable Atomistic
    Machine Learning},
    shorttitle = {Metatensor and Metatomic},
    author = {Bigi, Filippo and Abbott, Joseph W. and Loche, Philip and Mazitov, Arslan
    and Tisi, Davide and Langer, Marcel F. and Goscinski, Alexander and Pegolo, Paolo
    and Chong, Sanggyu and Goswami, Rohit and Chorna, Sofiia and Kellner, Matthias and
    Ceriotti, Michele and Fraux, Guillaume},
    year = {2025},
    month = aug,

>     number = {arXiv:2508.15704},
>     eprint = {2508.15704},
>     primaryclass = {physics},
>     publisher = {arXiv},
>     doi = {10.48550/arXiv.2508.15704},
>     urldate = {2025-08-22},
>     archiveprefix = {arXiv}

Reviewer: ```suggestion
    publisher = {arXiv},
    doi = {10.48550/arXiv.2508.15704},
```
